### DISCUSSION TITLE
Istio pilot support for topology aware routing with endpoint slice hints

### DISCUSSION THREAD

**Discussion by mmerrill3** [istio/istio#53054] [General]
Hi all, I see that the hints for topology aware routing (the availability zone hint) received from the endpoint slice controller when building the EDS list is not consumed.  Are there plans for leveraging this ability?  I am interested in ambient mode with layer 7 waypoint proxies deployed.  Given the endpoints that make up a given waypoint's service, if the waypoint service does satisfy the conditions for topology aware routing, the endpoint slices for the waypoint will contain the availability zone hint.  For ztunnel, are there plans to leverage these hints?

**Comment by howardjohn** [comment]
I had largely thought this was deprecated and replaced by https://kubernetes.io/docs/concepts/services-networking/service/#traffic-distribution. Currently, that is supported on ztunnel

**Reply by howardjohn** [reply]
Issue to add it to our envoy dataplanes: https://github.com/istio/istio/issues/53055. In the meantime, all of these support DestinationRule failover.

But for ztunnel -> waypoint, its always preferring local zone, even if trafficDistribution is not used.

**Reply by mmerrill3** [reply]
Thanks for the response.  Yes, Topology Aware Routing is being superseded by trafficDistribution.  I'm on k8s 1.29, so TAR is still there for now.  I was curious if I have a given service with a large amount of CPUs in use, spread across zones, where TAR is toggled on consistently.  Namely, I was curious what happens when the waypoint would be in a different AZ than the ztunnel on the node for the destination pod.  Given that ztunnel -> (dest. waypoint) will prefer local, I need to try to ensure waypoints actually do exist in the same AZ as the node for the destination pod.  Is the preference for ztunnel -> waypoint to be kept in the local zone defined in the ztunnel project, or is that in an EDS discovery library?

**Reply by howardjohn** [reply]
Defined in the Istiod control plane: https://github.com/istio/istio/blob/f5fcb30dd9058086378753ebbe6648e2d24d99ba/pilot/pkg/serviceregistry/kube/controller/ambient/services.go#L203

Likely we would just make the waypoint service default to having TrafficDistribution set, but its too new to depend on, so we hardcode the semantics.
